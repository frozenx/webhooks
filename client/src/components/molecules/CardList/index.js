import React, { Fragment } from 'react';
import AddressBookCard from '../../atoms/Card';
import { FormHeading, NotFoundContainer } from "../../../common/styles";
import i18Data from '../../../languagepack';
import Loader from '../../atoms/Loader';
import Label from '@beans/label';
import Button from '@beans/button';
import WithPagination from '../PaginationContainer';
import PaginationComponent from  '../Pagination';

const CardList = ({ addressList, navigateToForm, showLoader, handleBackComponent, selectedAttr }) => {

    return (
        <Fragment>
            {showLoader ? <Loader /> :
                <React.Fragment>
                     {addressList.length > 0 ? 
                        <FormHeading> <h2 dark={true} emphasized={true}>{`Site ${i18Data.addresses}`}</h2> </FormHeading> :
                        <NotFoundContainer>
                            <Label dark = {true} emphasized = {true}>{i18Data.noAddressFound}</Label> 
                            <Button variant = "secondary" onClick = {() => handleBackComponent(selectedAttr)}>{i18Data.cancel}</Button>
                        </NotFoundContainer>
                    }
                    {addressList.map(address => <AddressBookCard address={address} navigateToForm={navigateToForm} />)}
                </React.Fragment>
            }
        </Fragment>
    )
}

const CardListWithPagination = WithPagination(PaginationComponent, CardList)

export default CardListWithPagination;