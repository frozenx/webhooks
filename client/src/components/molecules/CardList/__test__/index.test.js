import React from 'react';
import CardList from '../'
import renderer from 'react-test-renderer'

describe('molecules/CardList', () => {
    describe('Given the CardList component is mounted', () => {
        test('it should render the markup' , () => {
            const tree = renderer.create(<CardList />).toJSON
            expect(tree).toMatchSnapshot()
        })
    })
})