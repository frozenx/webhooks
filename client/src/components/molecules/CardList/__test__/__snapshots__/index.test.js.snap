// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/CardList Given the CardList component is mounted it should render the markup 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HOC />,
        <ForwardRef>
          <h2>
            Site addresses
          </h2>
        </ForwardRef>,
        <Card />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h2>
            Site addresses
          </h2>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Site addresses",
          },
          "ref": null,
          "rendered": "Site addresses",
          "type": "h2",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "rules": Array [
              [Function],
              ";",
              "-webkit-tap-highlight-color:",
              [Function],
              ";border:0;box-sizing:border-box;margin:0;outline:",
              [Function],
              ";padding:0;position:relative;",
              ";",
              "
    .input-error {
        color: #cc3333;
    }
    h1,h2 {
        margin: 0 0 19px 0;
        color: #333;
    }
    h1,h2 {
        font-size: 32px;
    }
    h2{
        font-size: 28px;
    }
    h4 {
        margin: 0 0 20px 0;
        color: #666666;
        font-size: 16px;
        font-weight: normal;
    }
",
            ],
          },
          "displayName": "Styled(base-components__RootElement)",
          "foldedComponentIds": Array [
            "base-components__RootElement-sc-150tvch-1",
          ],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HOC />,
          <ForwardRef>
            <h2>
              Site addresses
            </h2>
          </ForwardRef>,
          <Card />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h2>
              Site addresses
            </h2>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Site addresses",
            },
            "ref": null,
            "rendered": "Site addresses",
            "type": "h2",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": false,
              "rules": Array [
                [Function],
                ";",
                "-webkit-tap-highlight-color:",
                [Function],
                ";border:0;box-sizing:border-box;margin:0;outline:",
                [Function],
                ";padding:0;position:relative;",
                ";",
                "
    .input-error {
        color: #cc3333;
    }
    h1,h2 {
        margin: 0 0 19px 0;
        color: #333;
    }
    h1,h2 {
        font-size: 32px;
    }
    h2{
        font-size: 28px;
    }
    h4 {
        margin: 0 0 20px 0;
        color: #666666;
        font-size: 16px;
        font-weight: normal;
    }
",
              ],
            },
            "displayName": "Styled(base-components__RootElement)",
            "foldedComponentIds": Array [
              "base-components__RootElement-sc-150tvch-1",
            ],
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
